uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/pageObjects/LoginPage.java

```diff
@@ -3,15 +3,15 @@
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 
-import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.matchesPattern;
 
 public class LoginPage extends Page {
 
     static final protected String urlPath = "/login";
 
     public LoginPage(WebDriver driver) {
         super(driver);
-        validateUrl(driver, endsWith(urlPath));
+        validateUrl(driver, matchesPattern(".*" + urlPath + "(\\?.*)?$"));
```

Comment:
I think that I would prefer

```
endsWith(urlPath + "(\\?.*)?")
```

is there a reason you moved away from `endsWith`?

Reply:
I needed to change the matcher when I found that the URL sometimes has GET parameters in it. `endsWith()` does not accept a regex, just a literal string. I chose to use a regex instead of `contains()` so the matcher can be a bit more precise with matching the start of the parameters.

Reply:
> I found that the URL sometimes has GET parameters in it

I would like it if the code could convey that information